target_chains/solana/programs/pyth-solana-receiver/src/lib.rs

```diff
@@ -0,0 +1,260 @@
+            .map_err(|_| ProgramError::InvalidArgument)?;
+        match accumulator_update_data.proof {
+            Proof::WormholeMerkle { vaa, updates: _ } => {
+                let vaa: Vaa<&RawMessage> = serde_wormhole::from_slice(vaa.as_ref()).unwrap();
+                let (header, body): (Header, Body<&RawMessage>) = vaa.into();
+                require_eq!(
+                    <wormhole_sdk::Chain as Into<u16>>::into(body.emitter_chain),
+                    emitter_chain,
+                    ReceiverError::InvalidEmitterChain
+                );
+                let post_vaa_ix_data = PostVAAInstructionData {
+                    version:            header.version,
+                    guardian_set_index: header.guardian_set_index,
+                    timestamp:          body.timestamp,
+                    nonce:              body.nonce,
+                    emitter_chain:      body.emitter_chain.into(),
+                    emitter_address:    body.emitter_address.0,
+                    sequence:           body.sequence,
+                    consistency_level:  body.consistency_level,
+                    payload:            body.payload.to_vec(),
+                };
+                let post_vaa_ix = Instruction {
+                    program_id: ctx.accounts.post_vaa_program.key(),
+                    accounts:   vec![
+                        AccountMeta::new_readonly(ctx.accounts.guardian_set.key(), false),
+                        AccountMeta::new_readonly(ctx.accounts.bridge_config.key(), false),
+                        AccountMeta::new_readonly(ctx.accounts.signature_set.key(), false),
+                        AccountMeta::new(ctx.accounts.vaa.key(), false),
+                        AccountMeta::new(ctx.accounts.payer.key(), true),
+                        AccountMeta::new_readonly(Clock::id(), false),
+                        AccountMeta::new_readonly(Rent::id(), false),
+                        AccountMeta::new_readonly(ctx.accounts.system_program.key(), false),
+                    ],
+                    data:       (POST_VAA, post_vaa_ix_data).try_to_vec()?,
+                };
+                invoke(&post_vaa_ix, ctx.accounts.to_account_infos().as_slice())?;
+            }
+        }
+        Ok(())
+    }
+
+    /// Verify the updates using the posted_vaa account
+    ///  * `vaa_hash` hash post of the post_vaa data to derive the address of the post_vaa account
+    ///  * `emitter_chain` expected emitter_chain from the post_vaa account
+    ///  * `price_updates` Vec of bytes for the updates to verify and post on-chain
+    ///
+    /// TODO:
+    ///    - use a `config` account that can only be modified by governance for checking emitter_chain
+    ///      and other constraints
+    ///
+
+    #[allow(unused_variables)]
+    pub fn post_updates(
+        ctx: Context<PostUpdates>,
+        vaa_hash: [u8; 32], // used for pda seeds
+        emitter_chain: u16,
```

Comment:
don't put a security critical parameter as an argument. this is just asking for us to forget about this TODO and leave a security vuln in the codebase. I would prefer you hardcoding a constant with the TODO above than this.

In general, I think we should strive to ensure that *every* PR leaves the code in a state that is OK from a security perspective. It may not be as configurable or flexible as we would like, or the functionality may not be there, but at least it's secure. That approach ensures that our own human fallibility doesn't cause security holes in our code.